src/mito2/src/read/seq_scan.rs

```diff
@@ -179,16 +227,50 @@ impl SeqScan {
                 }
             };
             if compat::has_same_columns(self.mapper.metadata(), reader.metadata()) {
-                builder.push_batch_reader(Box::new(reader));
+                sources.push(Source::Reader(Box::new(reader)));
             } else {
                 // They have different schema. We need to adapt the batch first so the
-                // mapper can convert the it.
+                // mapper can convert it.
                 let compat_reader =
                     CompatReader::new(&self.mapper, reader.metadata().clone(), reader)?;
-                builder.push_batch_reader(Box::new(compat_reader));
+                sources.push(Source::Reader(Box::new(compat_reader)));
             }
         }
-        Ok(Box::new(builder.build().await?))
+
+        Ok(sources)
+    }
+
+    /// Returns whether to use a parallel reader.
+    fn use_parallel_reader(&self) -> bool {
+        self.parallelism.allow_parallel_scan() && (self.files.len() + self.memtables.len()) > 1
+    }
+
+    /// Scan the input source in another task.
+    fn spawn_scan_task(&self, mut input: Source, semaphore: Arc<Semaphore>) -> BoxedBatchStream {
+        let (sender, receiver) = mpsc::channel(self.parallelism.channel_size);
+        tokio::spawn(async move {
```

Comment:
Should we use `comon_runtime`?

Reply:
Yes. I used to use read runtime but I changed it to current runtime finally. I'm not sure whether we should maintain so many runtimes.

Reply:
Some question, too. Does it seem helpful for profiling?

Reply:
Not always helpful. We might still need to use pprof.